conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+ * taking advantage of the superior observability and stability provided by Dialogue.
+ */
+final class DialogueFeignClient implements feign.Client {
+
+    private static final String PATH_TEMPLATE = "hr-path-template";
+    private static final Splitter pathSplitter = Splitter.on('/').omitEmptyStrings();
+    private static final Splitter.MapSplitter querySplitter =
+            Splitter.on('&').omitEmptyStrings().withKeyValueSeparator('=');
+
+    private final Clients clients;
+    private final Channel channel;
+    private final String baseUrl;
+    private final String serviceName;
+    private final String version;
+
+    DialogueFeignClient(Class<?> jaxrsInterface, Channel channel, ConjureRuntime runtime, String baseUrl) {
+        this.channel = Preconditions.checkNotNull(channel, "Channel is required");
+        this.serviceName = jaxrsInterface.getSimpleName();
+        this.version = Optional.ofNullable(jaxrsInterface.getPackage().getImplementationVersion())
+                .orElse("0.0.0");
+        this.baseUrl = baseUrl;
+        this.clients = runtime.clients();
+    }
+
+    @Override
+    public feign.Response execute(Request request, Request.Options _options) throws IOException {
+        com.palantir.dialogue.Request.Builder builder =
+                com.palantir.dialogue.Request.builder().body(requestBody(request));
+        request.headers().forEach((headerName, values) -> {
+            if (includeRequestHeader(headerName)) {
+                builder.putAllHeaderParams(headerName, values);
+            }
+        });
+
+        try {
+            Response response = clients.block(clients.call(
+                    channel, new FeignRequestEndpoint(request), builder.build(), IdentityDeserializer.INSTANCE));
+            return feign.Response.create(
+                    response.code(),
+                    null,
+                    Multimaps.asMap((Multimap<String, String>) response.headers()),
+                    new DialogueResponseBody(response));
+        } catch (UncheckedExecutionException e) {
+            // Rethrow IOException to match standard feign behavior
+            Throwable cause = e.getCause();
+            if (cause instanceof IOException) {
+                throw (IOException) cause;
+            }
+            throw e;
+        }
+    }
+
+    private static boolean includeRequestHeader(String headerName) {
+        // Content-type and content-length headers are handled by requestBody
+        if (HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        if (HttpHeaders.CONTENT_LENGTH.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        // Tracing path template is informational only
+        if (PATH_TEMPLATE.equalsIgnoreCase(headerName)) {
+            return false;
+        }
+        return true;
+    }
+
+    private static String urlDecode(String input) {
+        if (input.contains("%")) {
```

Comment:
I think the string `foo+bar` should also be urldecoded to `foo bar`. It seems like URLDecoder.decode already preserves its input in the case where there are no `+` or `%` chars, so probably safe to delete this if condition?

Reply:
Good catch!